### PR TITLE
Add nsxt_logical_switch data source

### PR DESCRIPTION
**Pull request by phanama**
This PR adds the nsxt_logical_switch data source. Our use case is to fetch logical switch ID by its name in terraform.

Requesting reviews and comments! @annakhm

### CODE REVIEW

**Comment by annakhm** [comment]
Hi @phanama, thanks for this effort! Please take a look at test errors CI is showing.
`go test -timeout=30s -parallel=4 github.com/terraform-providers/terraform-provider-nsxt github.com/terraform-providers/terraform-provider-nsxt/nsxt 
# github.com/terraform-providers/terraform-provider-nsxt/nsxt [github.com/terraform-providers/terraform-provider-nsxt/nsxt.test]
nsxt/data_source_nsxt_logical_switch_test.go:60:20: undefined: getTransportZoneID
?   	github.com/terraform-providers/terraform-provider-nsxt	[no test files]
FAIL	github.com/terraform-providers/terraform-provider-nsxt/nsxt [build failed]
make: *** [test] Error 123
The command "make test" exited with 2.`

**Comment by phanama** [comment]
@annakhm Looks like I skipped adding the utils_test.go. I've pushed it in the new commit, plus several missing changes on the data source's schema and testAcc case.

File: nsxt/data_source_nsxt_logical_switch_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				return fmt.Errorf("Unexpected status returned during logical switch deletion: %v", responseCode.StatusCode)
+			}
+			return nil
+		}
+	}
+	return fmt.Errorf("Error while deleting logical switch '%s': service not found", switchName)
+}
+
+func testAccNSXLogicalSwitchReadTemplate(name string) string {
+	return fmt.Sprintf(`
```
Comment (annakhm) [inline nsxt/data_source_nsxt_logical_switch_test.go:109]: May I suggest using logical switch resource to create the resource for this test? This way we won't need to introduce the new environment variable, and require less code.

Comment (phanama) [reply nsxt/data_source_nsxt_logical_switch_test.go:109]: That way, changes on the resource_nsxt_logical_switch might possibly affect the data source's test and in some cases possibly require some changes on the data source's code itself.
If that's alright or even preferable, we can do that. Wdyt?

Comment (annakhm) [reply nsxt/data_source_nsxt_logical_switch_test.go:109]: I think its OK. We prefer to have as little pre-configuring as possible before we run acc tests, so not having to pre-create the LS would help. Thanks!

Comment (phanama) [reply nsxt/data_source_nsxt_logical_switch_test.go:109]: Sorry for the really late reply!
When trying to do this, I can't find any example reference for using the resource in the data source test. Can you point me to any? All the other data sources seem to use the api directly to pre-create the related resource in the test.

**Pull request closed without merging**